src/Gaufrette/Adapter/AwsS3.php

```diff
@@ -111,10 +111,12 @@ public function rename($sourceKey, $targetKey)
 
         try {
             $this->service->copyObject($options);
-            return true;
         } catch (\Exception $e) {
             return false;
         }
+        $this->delete($sourceKey);
```

Comment:
Does Gaufrette have a clear idea about what it *should* do when delete fails - should this operation be atomic? (If not, then it probably should.)

The local adapter uses PHP's `rename` function which "attempts" to rename - unclear whether that operation is atomic. The v1 Amazon S3 adapter returns `false` if the copy operation succeeds but the delete fails (i.e. not atomic). Maybe this one should do the same as v1, but I think Gaufrette should be opinionated about what happens.